### DISCUSSION TITLE
How to understand the 'Last full scan time' of the scanner?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#18465] [Q&A] [answered]
## Version information:
minio: RELEASE.2023-10-07T15-07-38Z
scan-speed=default

## Quetion description
After uploading an object to Minio, BucketUsageInfo, or `dataUsageInfo` (obtained through `mc stat ALIAS/bucket`) is not updated in real time. 
How long does it take for the object to be included in the BucketUsageInfo.  Is it `last full scan time` ?

And is the `Last full scan time` equal to (time taken+cycle duration)*16?

The test steps are as follows.

## Test steps:
1. Upload over 20,000 objects using Warp, with an object size of 512 KB.
2. Execute mc admin scanner info ALIAS to view the scan information.
3. Every 1-3 minutes, use mc stat to check the BucketUsageInfo.

## Test results:
Scanner information:
```bash
[root@master-bec47-0 dx]# mc admin scanner info scanminio
+-------------------------------------------------------------------------------------------------------+
| Last full scan time:   0d0h15m; Estimated 2872.26/month                                               |
| Current cycle:         (between cycles)                                                               |
| Active drives:          0                                                                             |
| -------------------------------------- Last Minute Statistics --------------------------------------- |
| Objects Scanned:       2311 objects; Avg: 127.589µs; Rate: 3327840/day                                |
| Versions Scanned:      2311 versions; Avg: 10.895µs; Rate: 3327840/day                                |
| Versions Heal Checked: 0 versions; Avg: 0ms                                                           |
| Read Metadata:         2311 objects; Avg: 62.922µs, Size: 366 bytes/obj                               |
| ILM checks:            2311 versions; Avg: 1.926µs                                                    |
| Check Replication:     2311 versions; Avg: 2.013µs                                                    |
| Verify Deleted:        0 folders; Avg: 0ms                                                            |
| Yield:                 5.34s total; Avg: 2.311ms/obj                                                  |
+-------------------------------------------------------------------------------------------------------+
```
After 10 minutes of completing the put operation, the BucketUsageInfo still increases. This means that BucketUsageInfo has a delay of about 10min.

When viewing the scan information of the Minio server, the `Last full scan time` is 15 minutes, and the `Current cycle`  updates every minute.

The variable `dataUsageUpdateDirCycles = 16` means that it will traverse all folders every 16 cycles. The statistics of the `Last full scan time` are also related to 'const wantCycles = 16'.

**Comment by klauspost** [comment, marked as answer]
> How long does it take for the object to be included in the BucketUsageInfo. Is it `last full scan time`?

Yes.

> And is the Last full scan time equal to (time taken+cycle duration)*16?

Yes.

**Comment by Clara12062** [comment]
OK, I get it!
Thanks for your reply!
